app/controllers/checkouts_controller.rb

```diff
@@ -3,13 +3,26 @@ class CheckoutsController < ApplicationController
 
   def create
     @book = Book.find(params[:book_id])
-    checkout = Checkout.create(book: @book, user: current_user)
+    if CheckOutBook.process(book: @book, user: current_user)
+      flash[:success] = t('.successful_message')
+      redirect_to root_path
+    else
+      flash[:error] = t('.unsuccessful_message')
```

Comment:
This action will attempt to render the `create` view, by default. Should it redirect? Or `render :new` (if there's a form for this action -- I can't remember)?